client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/HotRodOperation.java

```diff
@@ -162,13 +161,22 @@ public boolean complete(T value) {
       return super.complete(value);
    }
 
-   public void scheduleTimeout(EventLoop eventLoop) {
+   public void scheduleTimeout(Channel channel) {
       assert timeoutFuture == null;
-      this.timeoutFuture = eventLoop.schedule(this, channelFactory.socketTimeout(), TimeUnit.MILLISECONDS);
+      this.timeoutFuture = channel.eventLoop().schedule(new TimeoutRunnable(channel), channelFactory.socketTimeout(), TimeUnit.MILLISECONDS);
```

Comment:
Meh, that means a new allocation in each operation :-/ I wonder how much mess would be storing the channel directly in the `HotRodOperation`.

Reply:
Done